nilearn/glm/tests/test_utils.py

```diff
@@ -32,11 +37,64 @@ def test_full_rank():
 
 def test_z_score():
     p = np.random.RandomState(42).uniform(size=10)
-    assert_array_almost_equal(norm.sf(z_score(p)), p)
+    one_p = p[::-1]
+    zval_sf = norm.isf(p)
+    zval_cdf = norm.ppf(one_p)
+    zval = np.zeros(p.size)
+    zval[zval_sf < 0] = zval_cdf[zval_sf < 0]
+    zval[zval_sf >= 0] = zval_sf[zval_sf >= 0]
+    assert_array_almost_equal(z_score(p, one_p), zval)
```

Comment:
Hum, this is a bit disturbing one_p is not 1-p ?

Reply:
This is definitely another mistake from my side. But I believe 1-p is not the solution, either. If it was, we would not need to solve this problem at the first place. 

In this test function, I have now replaced the random generation of p-values (values within the interval [0.0, 1.0]) by the random sampling of the standard Student’s t distribution. Then, this values were converted into p-value and "cdf-value" using the sf and the cdf, respectively.

I have done the exactly same thing---and written an additional test---for F-values, by randomly sampling the F-distribution.

Reply:
Also,  note that for the first test of the numerical precision, I replaced the original p-values by t-values which will return p-values with the survival function of the same order of magnitude.